pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -1574,12 +1586,24 @@ private void createPartitionMetadataProvider(String reason) {
   private void updateIngestionDelay(int indexedMessageCount) {
     if ((indexedMessageCount > 0) && (_lastRowMetadata != null)) {
       // Record Ingestion delay for this partition
-      _realtimeTableDataManager.updateIngestionDelay(_lastRowMetadata.getRecordIngestionTimeMs(),
-          _lastRowMetadata.getFirstStreamRecordIngestionTimeMs(),
-          _partitionGroupId);
+      updateIngestionDelay(_lastRowMetadata);
     }
   }
 
+  private void updateIngestionDelay(RowMetadata metadata) {
+    _realtimeTableDataManager.updateIngestionDelay(metadata.getRecordIngestionTimeMs(),
+        metadata.getFirstStreamRecordIngestionTimeMs(),
```

Comment:
Should we update the ingestion delay metric when any of `getRecordIngestionTimeMs` or `getFirstStreamRecordIngestionTimeMs` is not valid ? Looks like the default value for this is `Long.MIN_VALUE`

Reply:
This is a good point, but the checking belongs in the core method in the IngestionDelayTracker class. I have modified it in such a way we don't publish the metric that comes with negative values in the ingestion tracker so that no matter where we invoke update ingestion delay it will do the right thing. 

Now we support proper publishing of metrics if only ingestion delay is supported, or only end to end is supported or both are unsupported, or both are supported, or both are unsupported